### PR TITLE
typescript-angular: fix angular tests by using in-memory fake backend

### PR DESCRIPTION
**Pull request by macjohnny**
### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (5.3.0), `6.0.x`
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

**Comment by macjohnny** [comment]
cc @akehir @amakhrov

File: samples/client/petstore/typescript-angular-v7-provided-in-root/tests/default/src/test/api.spec.ts [hunk 1]
```diff
@@ -79,59 +72,45 @@ describe(`API (functionality)`, () => {
...
-      expect(createdPet.name).toEqual(newPet.name);
-    });
-
     it(`should get the pet data by id`, async(() => {
       const petService = TestBed.get(PetService);
-      return petService.getPetById(createdPet.id).subscribe(
-        result => expect(result.name).toEqual(newPet.name),
-        error => fail(`expected a result, not the error: ${error.message}`),
+      return petService.addPet(newPet).pipe(
+        switchMap((addedPet: Pet) => petService.getPetById(addedPet.id))
```
Comment (akehir) [inline samples/client/petstore/typescript-angular-v7-provided-in-root/tests/default/src/test/api.spec.ts:84]: I'd probably use mergeMap here, but for the test it does not really matter.

**Review by akehir — APPROVED** [review]
Looks great, thanks @macjohnny

**Comment by amakhrov** [comment]
Thanks for doing that @macjohnny !

**Pull request merged**